plugins/source/azure/CONTRIBUTING.md

```diff
@@ -1,114 +1,30 @@
 # Azure Source Plugin Contribution Guide
 
-Thanks for contributing to CloudQuery! You are awesome. This document serves as a guide for adding new services and resources to the Azure source plugin.
+## Overview
 
-There are two steps to adding a new Azure resource:
+This plugin uses code-generation heavily to speed up development and take advantage of the [Azure SDK](https://github.com/Azure/azure-sdk-for-go) which is in itself auto-generate (for the most part).
 
-1. [Generate interfaces for the Azure SDK function(s) that fetch the resource](#1-generate-interfaces-for-the-azure-sdk-functions-that-fetch-the-resource)
-2. [Add a code generation recipe](#2-add-a-code-generation-recipe)
+There are two steps of auto-generation:
 
-As a prerequisite, ensure that API calls to list/describe the desired resource are present in [the old Azure client](https://github.com/Azure/azure-sdk-for-go#client-previous-versions) or [the new one](https://github.com/Azure/azure-sdk-for-go#management-new-releases), and make note of:
+### Discovery and auto-generating recipes
 
-- to which Azure service the resource belongs
-- the schema of the returned object(s)
+The first problem with generating all azure tables is even discovering which packages are available and then generating the recipes under `codegen2/`. For this we use a multi-stage discovery and generation process which for the most part should run periodically to update code but developer would rarely change that.
 
-## 1. Generate Interfaces for the Azure SDK Function(s) that Fetch the Resource
+Process:
 
-### Before you Start
+1) `scripts/update_azure_subpackages.sh` will curl `https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk` and update `go.mod` with all relevant sub-packages and download those modules locally.
+2) `codegen0` will then parse the source code using go `ast` to find all root `NewXClients` which provides `NewListPager` function and will generate all recipes for `codegen1/recipes/*`.
```

Comment:
Yes, they have to be separated because one is dependened on go modules existing (`codegen1` and `codegen2`) while the other doesn't need anything `codegen0`. I also thought combining them but this wont work and will create bunch of compilation and build errors if something fails.

Reply:
it can still cause issues with compilation depending on what failed in which step. `codegen0` and `codegen1` will be rarely used by anyone apart from CI and myself. Developers should just use `codegen2`.

Reply:
OK, but do we need `codegen1/recipes` committed to the repository?

Reply:
Yes, because this way we can run it and see if anything changed. I don't see a reason not to commit it.